app/renderer/components/tabs/content/audioTabIcon.js

```diff
@@ -6,44 +6,66 @@ const React = require('react')
 const {StyleSheet, css} = require('aphrodite/no-important')
 
 // Components
-const ImmutableComponent = require('../../immutableComponent')
+const ReduxComponent = require('../../reduxComponent')
 const TabIcon = require('./tabIcon')
 
+// State
+const tabState = require('../../../../common/state/tabState')
+
+// Actions
+const windowActions = require('../../../../../js/actions/windowActions')
+
+// Utils
+const frameStateUtil = require('../../../../../js/state/frameStateUtil')
+
 // Styles
 const globalStyles = require('../../styles/global')
 const tabStyles = require('../../styles/tab')
 
-class AudioTabIcon extends ImmutableComponent {
-  get pageCanPlayAudio () {
-    return !!this.props.frame.get('audioPlaybackActive')
+class AudioTabIcon extends React.Component {
+  constructor (props) {
+    super(props)
+    this.toggleMute = this.toggleMute.bind(this)
   }
 
-  get shouldShowAudioIcon () {
-    // We switch to blue top bar for all breakpoints but default
-    return this.props.frame.get('breakpoint') === 'default'
+  get audioIcon () {
+    const isMuted = this.props.pageCanPlayAudio && !this.props.audioMuted
+
+    return isMuted
+      ? globalStyles.appIcons.volumeOn
+      : globalStyles.appIcons.volumeOff
   }
 
-  get mutedState () {
-    return this.pageCanPlayAudio && !!this.props.frame.get('audioMuted')
+  toggleMute (event) {
+    event.stopPropagation()
+    windowActions.setAudioMuted(this.props.frameKey, this.props.tabId, !this.props.audioMuted)
   }
 
-  get audioIcon () {
-    return !this.mutedState
-      ? globalStyles.appIcons.volumeOn
-      : globalStyles.appIcons.volumeOff
+  mergeProps (state, dispatchProps, ownProps) {
+    const currentWindow = state.get('currentWindow')
+    const frame = frameStateUtil.getFrameByKey(currentWindow, ownProps.frameKey)
+
+    const props = {}
+    // used in renderer
```

Comment:
Can you tell me a little bit more about this. I added this because this way you exactly know what is used in renderer and what in other functions in the component. Because of that changing and looking for thing should be easier then looking at a big list of variables.